llvm/utils/TableGen/SearchableTableEmitter.cpp

```diff
@@ -677,6 +713,22 @@ void SearchableTableEmitter::collectEnumEntries(
     for (auto [Idx, Entry] : enumerate(SavedEntries))
       Enum.Entries.try_emplace(Entry.first, Entry.second.Name, Idx);
   }
+
+  // If this is a scoped enum with custom type size, check if all entries can
+  // fit.
+  if (Enum.Scoped && Enum.ScopedSize != 0 &&
```

Comment:
should we be tracking the max value assigned to any enum entry and use that for the size check (instead of assuming they will always be from 0-N-1)?